packages/react-aria-components/stories/Popover.stories.tsx

```diff
@@ -433,3 +433,40 @@ export const PopoverTriggerWidthExample = () => (
     </Popover>
   </DialogTrigger>
 );
+
+export const PopoverChildElementAnimationExample = () => (
+  <DialogTrigger>
+    <Button>
+      Open popover
+    </Button>
+    <Popover
+      placement="bottom start"
+      style={{
+        zIndex: 5
+      }}>
+      <Dialog>
+        <div
+          className="content"
+          style={{
+            background: 'Canvas',
+            color: 'CanvasText',
+            border: '1px solid gray',
+            padding: 30,
+            transition: 'opacity 0.5s'
+          }}>
+          <style>
+            {`
+              .react-aria-Popover[data-exiting] .content,
```

Comment:
This fixes the issue for Safari from what I can tell.

```suggestion
              @starting-style {
                .react-aria-Popover .content {
                  opacity: 0;
                }
              }
              .react-aria-Popover[data-exiting] .content,
```

Reply:
Yeah I don't think this should be necessary. `data-entering` should be enough. If it isn't working, then that's a bug we should investigate.